osfclient/models/storage.py

```diff
@@ -35,6 +37,8 @@ def _update_attributes(self, storage):
         self._new_folder_url = self._get_attribute(storage,
                                                    'links', 'new_folder')
         self._new_file_url = self._get_attribute(storage, 'links', 'upload')
+        self.known_file_dict = dict()
```

Comment:
I'd call these `self.known_files` and `self.known_folders`. To avoid having a dict and a set, can you store them both in `self.known_paths = dict()` where for folders we use `None` as value?

Performance wise testing for membership in a `set` and a `dict` should be ~same.

Reply:
Yeah that does seem a little cleaner. Will do.